src/shared/ui/Modal/Modal.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useRef } from 'react';
+import { motion } from 'framer-motion';
+
+import { MODAL_MOTION } from '@/shared/constants/motion';
+
+import { Portal } from '../Portal';
+
+type Props = {
+  isOpen: boolean;
+  closeModal: () => void;
+  children: React.ReactNode;
+  mode?: 'wait' | 'sync' | 'popLayout';
+};
+
+export function Modal({ isOpen, closeModal, children, mode }: Props) {
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (modalRef.current && !modalRef.current.contains(e.target as Node)) {
+        closeModal();
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isOpen, closeModal]);
+
+  return (
+    <Portal isOpen={isOpen} mode={mode}>
+      <motion.div
+        {...MODAL_MOTION}
+        className="fixed inset-0 z-30 flex w-full items-center justify-center"
+      >
+        <div className="absolute inset-0 bg-black bg-opacity-50" onClick={closeModal} />
+
+        <div
+          ref={modalRef}
+          onClick={(e) => e.stopPropagation()}
+          className="relative z-40 items-center justify-center rounded-lg bg-white"
+        >
+          {children}
+        </div>
+      </motion.div>
+    </Portal>
```

Comment:
다음과 같이 사용이 가능할 것 같아요. 
`<div className="absolute inset-0 bg-black bg-opacity-50" onClick={closeModal} />` 해당 코드의 역할은 무엇인가요??
```suggestion
    <Portal isOpen={isOpen} mode={mode}>
      <motion.div
        {...MODAL_MOTION}
         onClick={onClickOutsideDefault}
        className="fixed inset-0 z-30 flex w-full items-center justify-center bg-black bg-opacity-50"
      >
        <div
          className="relative z-40 items-center justify-center rounded-lg bg-white"
        >
          {children}
        </div>
      </motion.div>
    </Portal>
```

Reply:
배경을 클릭하는 것과 애니메이션을 따로 관리하고자 기존코드로 작성하였습니다 !
`motion.div`에 `onClick`을 넣었을 경우에 모달만 눌러도 같이 모달이 닫혀서 배경과 요소를 분리하고자 다음과 같이 작성했습니다